src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -0,0 +1,92 @@
+  return validationSchema.validate(techRecordFields, {stripUnknown: true});
+};
+
+export const validateVehicleAttributes = (vehicleType: string, vehicleAttributes: any) => {
+  if (vehicleType !== VEHICLE_TYPE.TRL) {
+    return validateRecordCompleteness(psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, vehicleAttributes);
+  } else {
+    return validateRecordCompleteness(trlCoreMandatoryVehicleAttributes, vehicleAttributes);
+  }
+};
+
+export const validateCoreAndNonCoreMandatoryTechRecordAttributes = (vehicleType: string, techRecordWrapper: ITechRecordWrapper) => {
+  let coreMandatoryValidation;
+  let nonCoreMandatoryValidation;
+  const techRecord = techRecordWrapper.techRecord[0];
+  if (vehicleType === VEHICLE_TYPE.HGV) {
+    coreMandatoryValidation = validateRecordCompleteness(hgvCoreMandatorySchema, techRecord);
+    nonCoreMandatoryValidation = validateRecordCompleteness(hgvNonCoreMandatorySchema, techRecord);
+  } else if (vehicleType === VEHICLE_TYPE.PSV) {
+    coreMandatoryValidation = validateRecordCompleteness(psvCoreMandatorySchema, techRecord);
+    nonCoreMandatoryValidation = validateRecordCompleteness(psvNonCoreMandatorySchema, techRecord);
+  } else if (vehicleType === VEHICLE_TYPE.TRL) {
+    const trlRecord = {...techRecord, primaryVrm: techRecordWrapper.primaryVrm};
+    coreMandatoryValidation = validateRecordCompleteness(trlCoreMandatorySchema, trlRecord);
+    nonCoreMandatoryValidation = validateRecordCompleteness(trlNonCoreMandatorySchema, trlRecord);
+  } else if (vehicleType === VEHICLE_TYPE.LGV) {
+    coreMandatoryValidation = validateRecordCompleteness(lgvCoreMandatorySchema, techRecord);
+  } else if (vehicleType === VEHICLE_TYPE.CAR) {
+    coreMandatoryValidation = validateRecordCompleteness(carCoreMandatorySchema, techRecord);
+  } else if (vehicleType === VEHICLE_TYPE.MOTORCYCLE) {
+    coreMandatoryValidation = validateRecordCompleteness(motorcycleCoreMandatorySchema, techRecord);
+  }
+  return {coreMandatoryValidation, nonCoreMandatoryValidation};
+};
+
+export const computeRecordCompleteness = (techRecordWrapper: ITechRecordWrapper): string => {
+  let recordCompleteness = RECORD_COMPLETENESS_ENUM.COMPLETE;
+  let coreMandatoryCompleted = true;
+  let nonCoreMandatoryCompleted = true;
+  if (!techRecordWrapper.systemNumber) {
+    throw new HTTPError(400, ERRORS.SYSTEM_NUMBER_GENERATION_FAILED);
+  }
+  const vehicleAttributes = {
+    systemNumber: techRecordWrapper.systemNumber,
+    vin: techRecordWrapper.vin,
+    primaryVrm: techRecordWrapper.primaryVrm,
+    trailerId: techRecordWrapper.trailerId
+  };
+  const vehicleType = techRecordWrapper.techRecord[0].vehicleType;
+  if (!vehicleType) {
+    return RECORD_COMPLETENESS_ENUM.SKELETON;
+  }
+  const vehicleAttributeValidation = validateVehicleAttributes(vehicleType, vehicleAttributes);
+  if (vehicleAttributeValidation.error) {
+    coreMandatoryCompleted = false;
+  }
```

Comment:
lines can be reduced to

`isCoreMandatoryValid = !!validateVehicleAttributes(vehicleType, vehicleAttributes). error;`

Reply:
done